tests/fx/optimization/test_transformations.py

```diff
@@ -267,3 +269,105 @@ def test_change_truediv_to_mul_by_inverse():
     restored_num_mul = sum((1 if node.target == operator.mul else 0 for node in restored.graph.nodes))
 
     assert (orig_num_truediv == restored_num_truediv) and (orig_num_mul == restored_num_mul)
+
+
+def test_fuse_conv2d_batchnorm2d():
+    model = AutoModelForImageClassification.from_pretrained("fxmarty/resnet-tiny-beans")
+
+    traced_model = symbolic_trace(model, input_names=["pixel_values"], disable_check=True)
+
+    num_batchnorm2d = sum(1 if isinstance(mod, torch.nn.BatchNorm2d) else 0 for mod in traced_model.modules())
+    assert num_batchnorm2d != 0, "there should be at least one BatchNorm2d in the model to actually perform the test"
+
+    dummy_input = torch.rand(8, 3, 224, 224)
+    output_original = model(pixel_values=dummy_input)
+
+    transformation = FuseBatchNorm2dInConv2d()
+    transformed_model = transformation(traced_model)
+
+    num_batchnorm2d = sum(1 if isinstance(mod, torch.nn.BatchNorm2d) else 0 for mod in transformed_model.modules())
+    assert num_batchnorm2d == 0, "there should be no BatchNorm2d left in the model after the transformation"
+
+    output_transformed = transformed_model(pixel_values=dummy_input)
+
+    assert torch.allclose(output_original.logits, output_transformed["logits"], atol=1e-6)
+
+
+def test_fuse_linear_batchnorm1d():
+    config = GroupViTConfig()
+    model = GroupViTModel(config)
+    model.text_projection[1].weight.data = torch.rand(model.text_projection[1].weight.data.shape)
+    model.text_projection[1].bias.data = torch.rand(model.text_projection[1].bias.data.shape)
+    model.eval()
+
+    traced_model = symbolic_trace(
+        model, input_names=["input_ids", "attention_mask", "pixel_values"], disable_check=True
+    )
+
+    num_batchnorm1d = sum(1 if isinstance(mod, torch.nn.BatchNorm1d) else 0 for mod in traced_model.modules())
+    assert num_batchnorm1d != 0, "there should be at least one BatchNorm1d in the model to actually perform the test"
+
+    dummy_input = {
+        "pixel_values": torch.rand(1, 3, 224, 224, dtype=torch.float32),
+        "input_ids": torch.randint(low=0, high=100, size=(2, 32)),
+        "attention_mask": torch.ones(2, 32),
+    }
+    output_original = model(**dummy_input)
+
+    transformation = FuseBatchNorm1dInLinear()
+    transformed_model = transformation(traced_model)
+
+    num_batchnorm1d = sum(1 if isinstance(mod, torch.nn.BatchNorm1d) else 0 for mod in transformed_model.modules())
+    assert num_batchnorm1d == 0, "there should be no BatchNorm1d left in the model after the transformation"
+
+    output_transformed = transformed_model(**dummy_input)
+
+    assert torch.allclose(output_transformed["logits_per_image"], output_original.logits_per_image, atol=1e-6)
```

Comment:
Isn't it already checked in `_check_original_and_transformed_outputs_match`?

Reply:
Right, currently `_check_original_and_transformed_outputs_match` handles only NLP tasks, I will modify it to simplify the tests.